src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamespaceSymbol.cs

```diff
@@ -55,9 +55,15 @@ public sealed override ImmutableArray<Symbol> GetMembers()
         {
             EnsureAllMembersLoaded();
 
-            var builder = ArrayBuilder<Symbol>.GetInstance();
-            builder.AddRange(GetMemberTypesPrivate());
-            builder.AddRange(GetMemberNamespacesPrivate());
+            var memberTypes = GetMemberTypesPrivate();
+            var builder = ArrayBuilder<Symbol>.GetInstance(memberTypes.Length + lazyNamespaces.Count);
```

Comment:
If you are not planning to return the builder to the pool, you should just use `new`. 
- It will be cheaper than the same `new` after looking up in the pool. 
- It will keep long-lived builder instances in the pool, if there are any. On the other hand the array will be GC-ed, it is better if the instance is short-lived.

Reply:
💭 `ToImmutableAndFree()` returns the builder to the pool, right?

Reply:
Hmm. I thought it does not return. It actually returns an empty builder.

I guess, then it would be between benefits of reusing a builder object vs. possibly losing a tenured array to GC and contributing to eventual Gen2. 
It is not as straightforward to tell what is better.

Reply:
Actually I am curious - 2.2% savings - is that in "bytes allocated"?  what are the objects that we are saving? 

I assume you should end up allocating roughly the same number of arrays and the few that stay in the pool would not make a difference. And the builder objects get recycled the same - before the change and after.

Is the 2.2% difference just coming from the specifying the size upfront and not growing the builders?

Reply:
It still seems worth doing the change since you have the measurements. I just want to understand the change better.

Reply:
@VSadov The two cases that led to this change were the uses in `PENamespaceSymbol` and `CSharpSemanticModel`. PerfView suggested that these methods would obtain a builder, resize the array one or more times, and finally copy the result to a new array that was returned. Specifying the capacity up front results in at most one resize operation, and using `MoveToImmutable` eliminates the copy allocation in many cases. In all cases, the expected allocation overhead for code using this pattern is one array allocation of the actual size getting returned.